### ISSUE TITLE
Pylint treats tabs like spaces

### ISSUE THREAD

**Issue by sillypantscoder** [pylint-dev/pylint#9169]
### Bug description

```python
I use spaces instead of tabs. Pylint treats spaces as identical to tabs, and frequently reports that I am using one space, which is not a normal number.
```

### Command used

```shell
pylint file.py
```

### Pylint output

```python
************* Module baseHTTP2
file.py:9:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:10:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:11:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:12:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:15:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:16:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:17:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:18:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:21:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:22:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:23:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:26:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:27:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:28:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)
file.py:31:0: W0311: Bad indentation. Found 1 spaces, expected 4 (bad-indentation)

...etc...
```

### Expected behavior

I would expect all the "Bad indentation." notices to be gone.

### Pylint version

```shell
pylint 2.7.4
astroid 2.5.6
Python 3.9.6 (default, Jun 28 2021, 08:57:49) 
[GCC 10.3.0]
```

### OS / Environment

Pylint on VS Code for Linux. Also on Replit.

**Comment by Pierre-Sassoulas** [comment]
Do you mind upgrading pylint to see if the problem still exists in later version ? pylint 2.7.4 is very old, I initially thought we dropped bad-indentation, (like we dropped bad-continuation and bad-whitespace in pylint 2.6), but it's not the case.

**Comment by rogersheu** [comment]
@Pierre-Sassoulas 

Not sure if this fully addresses the original issue, but I tried changing my VSCode "Indent Using Spaces" setting to be from 4 to 1 for a file.

When I did that and ran pylint with no other changes, I got the `bad-indentation` the original issue brings up.

If I provide the following `.pylintrc` config, the message goes away.

```
[MASTER]
indent-string=" "
```

I have tested this on 3.3.0-dev0 and have not tested this on 2.7.4 (or older).

**Comment by Pierre-Sassoulas** [comment]
Thank you for the investigation @rogersheu. I wonder if using '\t' for the `indent-string` option on actual tab for indentation would fix the issue.
